src/vulkan/engine_vulkan.cc

```diff
@@ -159,108 +148,135 @@ Result EngineVulkan::Shutdown() {
 }
 
 Result EngineVulkan::CreatePipeline(amber::Pipeline* pipeline) {
-  const auto& engine_data = GetEngineData();
+  // Create the pipeline data early so we can access them as needed.
+  pipeline_map_[pipeline] = {};
+  auto& info = pipeline_map_[pipeline];
+
+  // Set VK_FORMAT_B8G8R8A8_UNORM for color frame buffer by default.
+  info.color_frame_format = MakeUnique<Format>();
+  info.color_frame_format->SetFormatType(kDefaultFramebufferFormat);
+  info.color_frame_format->AddComponent(FormatComponentType::kB,
+                                        FormatMode::kUNorm, 8);
+  info.color_frame_format->AddComponent(FormatComponentType::kG,
+                                        FormatMode::kUNorm, 8);
+  info.color_frame_format->AddComponent(FormatComponentType::kR,
+                                        FormatMode::kUNorm, 8);
+  info.color_frame_format->AddComponent(FormatComponentType::kA,
+                                        FormatMode::kUNorm, 8);
+
+  // Set VK_FORMAT_UNDEFINED for depth/stencil frame buffer in default.
+  info.depth_frame_format = MakeUnique<Format>();
+  info.depth_frame_format->SetFormatType(FormatType::kUnknown);
 
   // Handle Image and Depth buffers early so they are available when we create
   // the pipeline.
   for (const auto& info : pipeline->GetColorAttachments()) {
-    Result r = SetBuffer(info.type, static_cast<uint8_t>(info.location),
-                         info.buffer->AsFormatBuffer()->GetFormat(),
-                         info.buffer->GetData());
+    Result r = SetBuffer(
```

Comment:
This is confusing. Better not using `info` because we used it in `auto& info = pipeline_map_[pipeline];`